lib/ansible/plugins/lookup/env.py

```diff
@@ -32,21 +32,21 @@
     msg: "'{{ lookup('ansible.builtin.env', 'HOME') }}' is the HOME environment variable."
 
 - name: Before 2.13, how to set default value if the variable is not defined.
-        This cannot distinguish between USR undefined and USR=''.
```

Comment:
The original intent of the examples was to show the usage of the `default` filter, so the "typo" `USR` was intentional.

This should probably be change to something else, like `MISSING`, along with the rest of the example (`is the user`), or explicitly mention the variable is misspelled intentionally.

Reply:
if you test it with `$USER` you'll probably never get default as it is defined in 99% of the time, while `$USR` is very likely undefined.

Reply:
> But how does USER not show how to use the default filter, and USR do?

Because `USER` is defined on most systems, so it wouldn't trigger the `default` filter.

> And why not show a working example instead?

It is a working example of the `default` filter.

Reply:
The description of the examples in `name` of the tasks explicitly mention that they show setting a default value if the env variable is not defined, either by using the `default` filter or the `default` parameter of the `env` lookup.

If the examples use `USR` as the env variable name, it is likely that when users copy the example and run it they will see that the default value was correctly set because it is likely that `USR` isn't defined on the system, unlike `USER`.

Reply:
I'll revert back to `USR`, steal some text from your replies and update the doc.
How does that sound?

Reply:
the examples were added due to many requests for 'adding default to env lookup', normally a lookup only has examples for it's own usage w/o having to add 'commonly used filters' into them.